test/d_do_test.d

```diff
@@ -715,6 +747,17 @@ int tryMain(string[] args)
         }
     }
 
+    if(!autoCompileImports && testArgs.compiledImports.length > 0) {
+       autoCompileImports = true;
+       goto AGAIN;
+    }
```

Comment:
Why can't you do this beforehand (and avoid the `goto`)?

Reply:
huh?  This is supposed to cause the code to loop.  I could have wrapped the code in a loop but I wanted to avoid the extra level of nesting which would have made the diff a mess.